action_plugins/load_profile/__init__.py

```diff
@@ -0,0 +1,165 @@
+
+from PySide6 import QtCore
+from PySide6.QtCore import Property, Signal, Slot
+
+from gremlin import error, event_handler, input_devices, \
+    joystick_handling, mode_manager, shared_state, util
+from gremlin.base_classes import AbstractActionData, AbstractFunctor, \
+    DataCreationMode, Value
+from gremlin.config import Configuration
+from gremlin.profile import Library
+from gremlin.types import ActionProperty, AxisMode, InputType, PropertyType
+
+from gremlin.ui.action_model import SequenceIndex, ActionModel
+
+if TYPE_CHECKING:
+    from gremlin.ui.profile import InputItemBindingModel
+
+
+class LoadProfileFunctor(AbstractFunctor):
+
+    """Executes a load profile action callback."""
+
+    def __init__(self, action: LoadProfileData):
+        super().__init__(action)
+
+        self._profile_filename = action.profile_filename
+
+    def __call__(
+        self,
+        event: event_handler.Event,
+        value: Value
+    ) -> None:
+
+        if not self._should_execute(value):
+                return
+
+        logging.getLogger("system").debug(
+            f"Load profile ...{self._profile_filename}"
+        )
+
+class LoadProfileModel(ActionModel):
+
+    fileChanged = Signal()
+
+    def __init__(
+            self,
+            data: AbstractActionData,
+            binding_model: InputItemBindingModel,
+            action_index: SequenceIndex,
+            parent_index: SequenceIndex,
+            parent: QtCore.QObject
+    ):
+        super().__init__(data, binding_model, action_index, parent_index, parent)
+
+    def _qml_path_impl(self) -> str:
+        return "file:///" + QtCore.QFile(
+            "core_plugins:load_profile/LoadProfileAction.qml"
+        ).fileName()
+
+    def _icon_string_impl(self) -> str:
+        return LoadProfileData.icon
+
+    def _get_profile_filename(self) -> str:
+        return self._data.profile_filename
+
+    def _set_profile_filename(self, value: str) -> None:
+        if str(value) == self._data.profile_filename:
+            return
+        self._data.profile_filename = str(value)
+        self.fileChanged.emit()
+
+    profile_filename = Property(
+        str,
+        fget=_get_profile_filename,
+        fset=_set_profile_filename,
+        notify=fileChanged
+    )
+
+
+class LoadProfileData(AbstractActionData):
+
+    """Model of a load profile action."""
+
+    version = 1
+    name = "Load Profile"
+    tag = "load-profile"
+    icon = "\uF3B9"
+
+    functor = LoadProfileFunctor
+    model = LoadProfileModel
+
+    properties = [
+        ActionProperty.ActivateOnBoth
+    ]
+    input_types = [
+        InputType.JoystickAxis,
+        InputType.JoystickButton,
+        InputType.JoystickHat,
+        InputType.Keyboard
+    ]
+
+    def __init__(
+            self,
+            behavior_type: InputType=InputType.JoystickButton
+    ):
+        super().__init__(behavior_type)
+
+        # Model variables
+        self.profile_filename = ""
```

Comment:
Might be worth to use a `pathlib.Path` object here so you can easily verify the provided string is valid and possibly also use that for the valid aspect. The `is_valid()` bit is mainly there to avoid saving "bad" configuration bits.

Reply:
I implemented is_valid() using os.isfile and os.access. If you prefer pathlib.Path() then the profile_filename will become None at init and object when assigned. Let me know.

Reply:
I think both are fine, in most of the new parts of Gremlin I've starting using `Path` objects instead of strings and `os.` calls.

Reply:
Turns out pathlib.Path code lacks .access() to validate whether file is accessible. Left os methods since Path code turned out more lines and more complex.